src/ast/nodes/ExportDefaultDeclaration.ts

```diff
@@ -81,7 +83,14 @@ export default class ExportDefaultDeclaration extends NodeBase {
 				this.declaration.id === null,
 				options
 			);
-		} else if (this.variable.referencesOriginal()) {
+		} else if (
+			this.variable.referencesOriginal() &&
+			// Function and ClassExpresions only appear here if they were surrounded
+			// by parenthesis so no redeclaring will happen here
+			// export default (function a() {})
+			!isFunctionExpression(this.declaration) &&
+			!isClassExpression(this.declaration)
+		) {
```

Comment:
Looking through the code trying to understand the original cause I think the issue is that `.referencesOriginal()` should never be true for class and function expressions in the first place and I would rather fix the issue there (so that `.referencesOriginal()` does not "lie"). The problem is that in `bindNode()`, the presence of a `declarationName` is misinterpreted as this being a declaration. Then the call to `this.variable.setOriginalVariable()` triggers this unfortunate effect.

I see two possible approaches
1. Only set `declarationName` in `.initialiseNode()` if this is actually a class or function declaration or an identifier. This has the disadvantage that we do not get the nice name for *expressions.
2. Check in `.bindNode()` if we have a declaration or identifier and call `setOriginal()` only in those cases. I would prefer this approach.

Reply:
Okay i changed it to use approach 2.